src/main/webapp/scripts/private-room-script.js

```diff
@@ -1,52 +1,198 @@
-	event.target.playVideo();
+    document.getElementById('player-div').style.borderColor = '#FF6D00';
+    loadRoomPlaylist();
 }
 
-// The API calls this function when the player's state changes.
-// The function indicates that when playing a video (state=1),
-// the player should play for six seconds and then stop.
-function onPlayerStateChange(event) {
-	if (event.data == YT.PlayerState.PLAYING && !isVideoFinished) {
-		setTimeout(stopVideo, 6000);
-		isVideoFinished = true;
-	}
+function stopVideo() {
+    youtubePlayer.stopVideo();
 }
 
-function stopVideo() {
-	youtubePlayer.stopVideo();
+// Log state changes
+function onStateChange(event) {
+    var state = "undefiend";
+    switch (event.data) {
+        case YT.PlayerState.UNSTARTED:
+            state= "unstarted";
+            break;
+        case YT.PlayerState.ENDED:
+            state = "ended";
+            break;
+        case YT.PlayerState.PLAYING:
+            state = "playing";
+            break;
+        case YT.PlayerState.PAUSED:
+            state = "paused";
+            break;
+        case YT.PlayerState.BUFFERING:
+            state = "buffering";
+            break;
+        case YT.PlayerState.CUED:
+            state = "video cued";
+            break;
+        default:
+            state = "unknown (" + event.data + ")";
+    }
+    console.log('onStateChange: ' + state);
+}
+
+async function displayChat() {
+    let response = await fetch('/chat');
+    let messages = await response.json();
+    const messageElement = document.getElementById('chat-messages');
+    messageElement.innerHTML = '';
+    for (message in messages) {
+      commentElement.appendChild(createParagraph(comments[comment]));
+    }  
 }
+
+function createParagraph(msgJson) {
+    var msg = JSON.parse(msgJson);
+    const paragraph = document.createElement('p');
+    paragraph.innerHTML = '<h4>';
+    paragraph.innerText = msg.sender;
+    paragraph.innerHTML = '</h4>';
+    paragraph.innerText = msg.message;
+    return paragraph;
+}
+
+// Shows and refreshes the messages shown on the private room page
+async function displayChat() {    
+    let response = await fetch(`/chat?roomID=${window.roomId}`);
+    let messages = await response.json();    
+    const messageElement = document.getElementById('chat-messages');    
+    messageElement.innerHTML = '';    
+    for (message in messages) {      
+        messageElement.appendChild(createNewMessage(messages[message]));    
+    }      
+    displayChat();
+}
+
+// Inserts messages as HTML elements
+function createNewMessage(msg) {    
+    const listItem = document.createElement('li');    
+    listItem.innerHTML += '<div class="msgText"><p><span class="sender">'+msg.sender+': </span><span class="msgBody">'+msg.message+'</span></p> <span class="sub-text"> Sent at ' + toTime(msg.timestamp)+ '</span></div>';    
+    return listItem;
+}
+
+// Manipulates timestamp value to be displayed in hh:mm(+AM/PM) format 
+function toTime(ms) {    
```

Comment:
this definitely needs a unit test.

As discussed, create a secondary javascript file that will test this file.

function toTime_caseTested_expectedResult() {
  Set up ms variable

  run toTime function

  compare results
}

Reply:
Done